pkg/scanners/helm/parser/parser.go

```diff
@@ -0,0 +1,227 @@
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Parser) addPaths(ctx context.Context, paths ...string) error {
+	for _, path := range paths {
+		if _, err := fs.Stat(p.workingFS, path); err != nil {
+			return err
+		}
+
+		if strings.HasSuffix(path, "Chart.yaml") && p.rootPath == "" {
+			p.rootPath = filepath.Dir(path)
+		}
+		p.filepaths = append(p.filepaths, path)
+	}
+	return nil
+}
+
+func (p *Parser) RenderedChartFiles() ([]ChartFile, error) {
+
+	tempDir, err := ioutil.TempDir(os.TempDir(), p.chartName)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := p.writeBuildFiles(tempDir); err != nil {
+		return nil, err
+	}
+
+	workingChart, err := loadChart(tempDir)
+	if err != nil {
+		return nil, err
+	}
+
+	workingRelease, err := p.getRelease(workingChart)
+	if err != nil {
+		return nil, err
+	}
+
+	var manifests bytes.Buffer
+	_, _ = fmt.Fprintln(&manifests, strings.TrimSpace(workingRelease.Manifest))
+
+	splitManifests := releaseutil.SplitManifests(manifests.String())
+	manifestsKeys := make([]string, 0, len(splitManifests))
+	for k := range splitManifests {
+		manifestsKeys = append(manifestsKeys, k)
+	}
+	return p.getRenderedManifests(manifestsKeys, splitManifests), nil
+}
+
+func (p *Parser) getRelease(chart *chart.Chart) (*release.Release, error) {
+
+	r, err := p.helmClient.RunWithContext(context.Background(), chart, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if r == nil {
+		return nil, fmt.Errorf("there is nothing in the r")
+	}
+	return r, nil
+}
+
+func loadChart(tempFs string) (*chart.Chart, error) {
+	loadedChart, err := loader.Load(tempFs)
+	if err != nil {
+		return nil, err
+	}
+
+	if req := loadedChart.Metadata.Dependencies; req != nil {
+		if err := action.CheckDependencies(loadedChart, req); err != nil {
+			return nil, err
+		}
+	}
+	return loadedChart, nil
+}
+
+func (*Parser) getRenderedManifests(manifestsKeys []string, splitManifests map[string]string) []ChartFile {
+	sort.Sort(releaseutil.BySplitManifestsOrder(manifestsKeys))
+	var manifestsToRender []ChartFile
+	for _, manifestKey := range manifestsKeys {
+		manifest := splitManifests[manifestKey]
+		submatch := manifestNameRegex.FindStringSubmatch(manifest)
+		if len(submatch) == 0 {
+			continue
+		}
+		manifestsToRender = append(manifestsToRender, ChartFile{
+			TemplateFilePath: getManifestPath(manifest),
+			ManifestContent:  manifest,
+		})
+	}
+	return manifestsToRender
+}
+
+func getManifestPath(manifest string) string {
+	lines := strings.Split(manifest, "\n")
+	if len(lines) == 0 {
+		return "unknown.yaml"
+	}
+	return strings.TrimPrefix(lines[0], "# Source: ")
+}
+
+func (p *Parser) writeBuildFiles(tempFs string) error {
+	for _, path := range p.filepaths {
+		content, err := fs.ReadFile(p.workingFS, path)
+		if err != nil {
+			return err
+		}
+		workingPath := strings.TrimPrefix(path, p.rootPath)
+		workingPath = filepath.Join(tempFs, workingPath)
+		if err := os.MkdirAll(filepath.Dir(workingPath), os.ModePerm); err != nil {
```

Comment:
Should this be writing to a temporary filesystem instead of using `os.*`? Otherwise it'll write files into a project dir unless I'm misunderstanding (which is likely.)

Reply:
No, this is writing to the physical on disk temp location that will ultimately go into the helm library